File: 2015/02/02.py
with open('./input.txt', 'r') as file:
    problem_input = [s.strip() for s in file.readlines()]

def calculate_wrapping_paper(x, y, z):
    x, y, z = int(x), int(y), int(z)
    surfaces = [x*y, y*z, z*x]
    return  2*sum(surfaces) + min(surfaces)

print(sum([calculate_wrapping_paper(*el.split('x')) for el in problem_input]))

# 14.1 ms ±  4.9 ms (1000 runs, 100 warmup) - Python 3.10.5 (conda-forge)

File: 2015/02/02.js
const fs = require('fs');

var problemInput = fs.readFileSync('./input.txt', 'utf8').split('\n');

function calculateWrappingPaper(numStr) {
  numStr = numStr.split('x').map(str => parseInt(str))
  let x, y, z;
  [x, y, z] = [...numStr]
  let surface = [x*y, y*z, z*x]
  return  2*surface.reduce((a, b) => (a + b)) + Math.min(...surface)
}

let result = problemInput.map(calculateWrappingPaper);
console.log(result.reduce((a, b) => (a + b)));

// 84.6 ms ± 28.0 ms (1000 runs, 100 warmup) - Node.js 18.6.0 (conda-forge)